content/blog/2018/07/2018-07-12-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,190 @@
+  # default value will be false
+  pushPrOnSuccess: false
+
+  prApprovers:
+    - username1
+    - username2
+    - username3
+
+# Study about the reports type. Urgent
+reports:
+  - location_of_report_1
+  - location_of_report_2
+
+environment:
+  variables:
+    variable_1: value_1
+    variable_2: value_2
+
+  # Credentials contains only two fields. credentialId must be present in the Jenkins Credentials
+  credentials:
+    - credentialId : fileCredentialId
+      variable : FILE
+
+      # Username will be accessed by LOGIN_USR and Password will be accessed by LOGIN_PSW
+    - credentialId : dummyGitRepo
+      variable : LOGIN
+
+stages:
+  - name: satge1
+    agent: any
+    steps:
+      - sh: "scripts/hello"
+      - sleep:
+          time: 2
+          unit: SECONDS
+      - sleep: 2
+      - junit:
+          testResults: "target/**.xml"
+          allowEmptyResults: true
+          testDataPublishers:
+            - AutomateTestDataPublisher
+            - JunitResultPublisher:
+                urlOverride: "urlOverride"
+    # Post section for "stage1". All Conditions which are available in Jenkins
+    # declarative pipeline are supported
+    post:
+      failure:
+        - sh: "scripts/hello"
+
+# Outer post section. Just like declarative pipeline.
+post:
+  always:
+    - sh: "scripts/hello"
+----
+
+=== Coding Phase 2 plans
+
+* [*] Decide a proper YAML format to use for Jenkinsfile.yaml
+* [*] Create Step Configurator for SPRP plugin. https://issues.jenkins-ci.org/browse/JENKINS-51637[Jenkins-51637].
+This will enable users to use Pipeline steps in Jenkinsfile.yaml.
+* [*] Automatic indentation generation in the generated Pipeline SnipperGenerator class.
+* [*] Write tests for the plugin.
+
+=== Coding Phase 3 plans
+
+. Support for webhooks.
+. Mechanism to check if pull request is approved by trusted users.
+. Support report types.
+. Conditions like when.
+. Prepare documentation.
+
+
+https://issues.jenkins-ci.org/browse/JENKINS-52452[Phase 3 Jira Epic]
+
+
+=== How to reach me
+
+* Email: gautamabhishek46@gmail.com
+* Gitter room: link:https://gitter.im/Jenkinsci/simple-pull-request-job-plugin[]
+
+=== References
+
+* link:https://docs.google.com/document/d/1cuC0AvQG3e4GCjIoCwK3J0tcJVAz1eNDKV8d_zXxlO8/edit[Initial proposal of the project]
+* link:https://github.com/Jenkinsci/simple-pull-request-job-plugin[Project repository]
+* link:https://Jenkins.io/projects/gsoc/2018/simple-pull-request-job-plugin/[Project page]
+* link:https://gitter.im/Jenkinsci/simple-pull-request-job-plugin?utm_source=share-link&utm_medium=link&utm_campaign=share-link[Gitter chat]
+* link:https://issues.Jenkins-ci.org/issues/?jql=project%20%3D%20Jenkins%20AND%20component%20%3D%20simple-pull-request-job-plugin[Bug Tracker]
+* link:https://github.com/gautamabhishek46/dummy[Demo Repository]
+* link:https://www.youtube.com/watch?v=qWHM8S0fzUw[Phase 2 Presentation video](June 14, 2018)
```

Comment:
Phase 2 -> Phase 1

Reply:
I will paste the links of Phase 2 presentation slides and video